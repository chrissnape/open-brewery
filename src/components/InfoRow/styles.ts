import { StyleSheet } from 'react-native';
import { primaryColour, fontSize } from '../../utils/styles';

export default StyleSheet.create({
  keyLabel: {
    color: primaryColour,
    flex: 1,
    fontSize: fontSize,
    fontWeight: 'bold',
  },
  label: {
    flex: 2,
    fontSize: fontSize,
  },
  row: {
    flexDirection: 'row',
  },
});
