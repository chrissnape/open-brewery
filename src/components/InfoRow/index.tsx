import React, { FC } from 'react';
import { Text, TouchableWithoutFeedback, View } from 'react-native';
import styles from './styles';

type Props = {
  keyLabel: string;
  label: string;
  onPress?: () => void;
}

const BreweryCardComponent: FC <Props> = ({ keyLabel, label, onPress }): JSX.Element => (
  <View style={styles.row}>
    <Text style={styles.keyLabel}>{keyLabel}</Text>
    <TouchableWithoutFeedback onPress={onPress}>
      <Text style={styles.label}>{label}</Text>
    </TouchableWithoutFeedback>
  </View>
);

export default BreweryCardComponent;
