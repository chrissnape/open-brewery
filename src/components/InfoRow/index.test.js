import React from 'react';
import renderer from 'react-test-renderer';
import InfoRow from './index';

const props = {
  keyLabel: 'Lorem',
  label: 'Ipsum',
}

describe('<InfoRow />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<InfoRow { ...props } />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
