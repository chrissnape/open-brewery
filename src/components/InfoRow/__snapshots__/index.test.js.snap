// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoRow /> renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#065535",
        "flex": 1,
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    Lorem
  </Text>
  <Text
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 2,
        "fontSize": 16,
      }
    }
  >
    Ipsum
  </Text>
</View>
`;
